### PR TITLE
Add repo owner to doc source links

### PR DESCRIPTION
**Pull request by nateraw**
While adding some reference docs to `timm`'s Docs, I noticed the "Source" links were incorrect in the API documentation - they always pointed to `github.com/huggingface/timm`. We want them to point to `github.com/rwightman/pytorch-image-models`. 

This PR aims to make that happen. 

As of the time of creating this PR, there is an ambiguity between `package` and `package_name`, which still makes links generate incorrectly with `package_name=timm`, as the github repo name should still be `pytorch-image-models`. Will sort out a fix for that...

### CODE REVIEW

**Comment by nateraw** [comment]
@mishig25 do you think I should go with:

1. the route I'm doing here, where I'm parametrizing the repo name/repo owner (probably the right way to do it but I'll have to rewrite some tests and I'm afraid something in another repo will break)
2. Just adding explicit override when package name is `pytorch-image-models` to update github links/package name

**Comment by nateraw** [comment]
@sgugger mind having a look at this and weighing in on the above?

File: .github/workflows/build_pr_documentation.yml
Comment (sgugger) [inline .github/workflows/build_pr_documentation.yml:138]: Note that you will also need to change the command in the `build_documentation` job.

File: src/doc_builder/autodoc.py
Comment (sgugger) [inline src/doc_builder/autodoc.py:332]: Can we use a get here? not sure the `page_info` will always have everything.
So try `repo_name`, then default to `package_name` if not passed.

File: src/doc_builder/external.py
Comment (sgugger) [inline src/doc_builder/external.py:66]: Let's not rely on magic bool conversion :-)
```suggestion
    repo_name = repo_name if repo_name is not None else package_name
```

File: src/doc_builder/external.py
Comment (sgugger) [inline src/doc_builder/external.py:97]:
```suggestion
    repo_name = repo_name if repo_name is not None else package_name
```

File: src/doc_builder/external.py
Comment (sgugger) [inline src/doc_builder/external.py:104]: Are we sure that we want the doc to be hosted in the `repo_name` env? I'd expect hf.co/docs/timm, not hf.co/docs/pytorch-pretrained-models

**Review by sgugger — COMMENTED** [review]
Thanks for working on this. I think it's better to parametrize everything, just in case we end up re-using the doc-builder for other repos outside of huggingface, like bigscience (and since you have done most of the work already ;-) )

File: src/doc_builder/external.py
Comment (nateraw) [reply src/doc_builder/external.py:104]: good catch!!! Reading back through my logic here, I think this was a mistake

**Comment by nateraw** [comment]
@sgugger, I _think_ we're good to go here 🤞 I made updates to reflect your review comments (thanks so much!!)

I'll need to remove hard coded references to this PR's branch in the GitHub actions before we merge, though... needed to add that so I could work on https://github.com/rwightman/pytorch-image-models/pull/1575.

File: .github/workflows/build_pr_documentation.yml
Comment (mishig25) [inline .github/workflows/build_pr_documentation.yml:53]: this is correct way to test doc-build action changes 👍

**Comment by mishig25** [comment]
@nateraw, if I go to this [preview page](https://moon-ci-docs.huggingface.co/docs/timm/pr_1575/en/reference/optimizers) & click on one of the source links, for example: 
https://github.com/rwightman/pytorch-image-models/blob/vr_1575/src/timm/optim/optim_factory.py#L176

it should **not** have, string `src` & just be https://github.com/rwightman/pytorch-image-models/blob/vr_1575/timm/optim/optim_factory.py#L176

if it does not have `src` & if you replace manually `vr_1575` with `main`, the link works

**Review by sgugger — APPROVED** [review]
LGTM, once @mishig25 's bug is fixed of course :-)

**Comment by nateraw** [comment]
@mishig25 nice catch! I added the `version_tag_suffix` build arg to the workflows so I can pass it through from `timm`. Links look good now in that preview link you shared 😄 .

**Review by mishig25 — APPROVED** [review]
lgtm !

**Pull request merged**
